keras_core/layers/preprocessing/random_translation.py

```diff
@@ -73,43 +90,157 @@ def __init__(
         interpolation="bilinear",
         seed=None,
         fill_value=0.0,
-        name=None,
+        data_format=None,
         **kwargs,
     ):
-        if not tf.available:
-            raise ImportError(
-                "Layer RandomTranslation requires TensorFlow. "
-                "Install it via `pip install tensorflow`."
-            )
-
-        super().__init__(name=name)
-        self.layer = tf.keras.layers.RandomTranslation(
-            height_factor=height_factor,
-            width_factor=width_factor,
-            fill_mode=fill_mode,
-            interpolation=interpolation,
-            seed=seed,
-            fill_value=fill_value,
-            name=name,
-            **kwargs,
+        super().__init__(**kwargs)
+        self.height_factor = height_factor
+        self.height_lower, self.height_upper = self._set_factor(
+            height_factor, "height_factor"
         )
+        self.width_factor = width_factor
+        self.width_lower, self.width_upper = self._set_factor(
+            width_factor, "width_factor"
+        )
+        self._check_fill_mode_and_interpolation(fill_mode, interpolation)
```

Comment:
You can inline this logic here, no need for a separate method since it's only used 1x

Reply:
Done